### DISCUSSION TITLE
Tab Containers

### DISCUSSION THREAD

**Discussion by madprops** [madprops/grasshopper#163] [General]
I'm starting work on tab containers, which is a feature firefox has. I'll document progress here.

**Comment by madprops** [comment]
Containers will be represented by a regular item icon, which can be controlled like the rest of the icons.

It will represent the container color and it will be a square:

![container234](https://github.com/user-attachments/assets/7ed056fb-8b1b-4c46-965f-87beece9e5b2)

![contain34](https://github.com/user-attachments/assets/f02f82ec-3bff-4d0e-aaee-86238b41da9c)

Added 2 new commands:

![containcmd](https://github.com/user-attachments/assets/a44da193-5fdd-4a6e-8684-9de7614e7dac)

**Comment by madprops** [comment]
Adding an optional mode to show text alongside the icons:

![cotnainer234](https://github.com/user-attachments/assets/f4542ea1-17ca-4586-82c7-243c95dfc85d)

Adding more filters.

Adding it to the tab box.

**Comment by N3C2L** [comment]
Thanks for the containers! Wanted to try it out now, but after updating GH the tab list is completly blank. The tabs are not deleted, because in other tab extensions the tabs are visible. Even after restarting the browser or restarting only GH, tabs mode is blank. Bookmarks Mode or History Modes or Closed Modes all working, but Tabs Mode is blank.

**Reply by madprops** [reply]
If you open the global console with ctrl+shift+j do you see errors when you try to open it?

Maybe floorp doesn't support containers? (though it shouldn't crash)

I might have to make this an optional permission.

The console error would be the most helpful.

Just tried on a new floorp, it works ok, something in your setup must be causing a crash, I wonder what

**Reply by N3C2L** [reply]
Oh, sorry for the late reply, it seems that this thread discussion wasn't updating inside the github interface but now I can see the reply. I will look into my settings.

**Reply by madprops** [reply]
I just updated a new version with a setting in general to turn off container support, you could try that

But the best would be to open the global firefox console with ctrl+shift+j and then opening the sidebar/popup and see what errors pop up.

**Reply by N3C2L** [reply]
Thanks, I will try both. And yes, it seems it's something with my floorp configuration because you tested it on a fresh floorp, working + I tested it on another firefox version I have installed here on my pc, it's also working.

**Comment by madprops** [comment]
I'll check back tomorrow as I'm going to sleep

**Reply by N3C2L** [reply]
Alright, no problem. 👍

**Comment by N3C2L** [comment]
Had to to do an "Icons" settings category reset, then the tabs were visible again.  👍

And I discovered 2 problems from this icons category, here are the steps:

Settings > Icons> Icon Order > Clear the data here (I ordered these icons before the update so maybe this was the reason why the tabs were blank on my side)
Result: Tab list is blank after resuming from the settings to the tab list

Settings > Icons > Show Container Icons > Switch to "Never" (I have them on "Always", but I was curios to test it with this setting)
Result: Tab list is blank after resuming from the settings to the tab list

Regarding the containers:
I love them, thanks for doing that.
Maybe some enhancements (both taken from sideberys right click on tab menu):
1. Right click on a tab and cmd to reopen the tab in a specific container
2. Be able to create containers inside GH

**Reply by madprops** [reply]
The icon order of course! Your setting was modified and thus not updated, I should have increased the version number of the setting. Thanks for finding that out.

Alright adding "Open In Container" will require an extra "cookies" permission.

I won't add container creation for now, I think that's better suited to be managed by the browser, especially if it has to pick icons/colors etc.

https://addons.mozilla.org/en-US/firefox/addon/multi-account-containers

**Reply by N3C2L** [reply]
Thanks for the fixes and the "Open in Container" cmd! You are right, the container creation isn't important to do inside GH because it's not a thing that someone needs to do daily. They can be easily created in the browser settings or other addons.
